include/aws_iot_mqtt_client_interface.h

```diff
@@ -58,154 +58,242 @@ extern "C" {
+ * @functionpage{aws_iot_mqtt_attempt_reconnect,mqtt,attempt_reconnect}
+ */
 
 /**
- * @brief MQTT Client Initialization Function
+ * @brief Initialize a new MQTT client context.
  *
- * Called to initialize the MQTT Client
+ * This function should be called before any other MQTT function to initialize
+ * a new MQTT client context. Once the client context is no longer needed,
+ * @ref mqtt_function_free should be called.
  *
- * @param pClient Reference to the IoT Client
- * @param pInitParams Pointer to MQTT connection parameters
+ * @param[in] pClient MQTT client context to initialize
+ * @param[in] pInitParams Pointer to MQTT connection parameters
  *
- * @return IoT_Error_t Type defining successful/failed API call
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
  */
+/* @[declare_mqtt_init] */
 IoT_Error_t aws_iot_mqtt_init(AWS_IoT_Client *pClient, IoT_Client_Init_Params *pInitParams);
+/* @[declare_mqtt_init] */
+
+/**
+ * @brief Clean up an MQTT client context that is no longer needed.
+ *
+ * This function will free up resources used by an MQTT client context. It should
+ * only be called when that context is no longer needed.
+ *
+ * @param[in] pClient MQTT client context that was previously initialized by
+ * @ref mqtt_function_init
+ *
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
+ */
+/* @[declare_mqtt_free] */
+IoT_Error_t aws_iot_mqtt_free( AWS_IoT_Client *pClient );
+/* @[declare_mqtt_free] */
 
 /**
- * @brief MQTT Connection Function
+ * @brief Establish a connection with an MQTT server.
  *
- * Called to establish an MQTT connection with the AWS IoT Service
+ * This function should be called once and after @ref mqtt_function_init. It sends
+ * the MQTT CONNECT packet to the server, which establishes an MQTT session. Once
+ * the session is no longer needed, it can be closed with @ref mqtt_function_disconnect.
  *
- * @param pClient Reference to the IoT Client
- * @param pConnectParams Pointer to MQTT connection parameters
+ * @param[in] pClient MQTT client context
+ * @param[in] pConnectParams MQTT connection parameters
  *
- * @return An IoT Error Type defining successful/failed connection
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
  */
+/* @[declare_mqtt_connect] */
 IoT_Error_t aws_iot_mqtt_connect(AWS_IoT_Client *pClient, IoT_Client_Connect_Params *pConnectParams);
+/* @[declare_mqtt_connect] */
 
 /**
- * @brief Publish an MQTT message on a topic
+ * @brief Publish an MQTT message to a topic.
  *
- * Called to publish an MQTT message on a topic.
- * @note Call is blocking.  In the case of a QoS 0 message the function returns
- * after the message was successfully passed to the TLS layer.  In the case of QoS 1
- * the function returns after the receipt of the PUBACK control packet.
+ * This function sends an MQTT message to the server. The server will then
+ * forward this message to any clients with subscriptions to topic filters
+ * that match the message's topic.
```

Comment:
"_The server will then forward this message to any clients with subscriptions to topic filters that match the message's topic_" - This is not necessary to be true for every broker (or event for AWS IoT), is it?

Reply:
Please clarify.  Servers compliant with MQTT spec should forward messages to clients with matching topic subscriptions.